<?xml version="1.0" encoding="UTF-8"?>
<launch>

    <arg name="icp_config_name" default="icp_l515.yaml" />
    <arg name="icp_config_path" default="$(find map_registration)/params/$(arg icp_config_name)" />

    <arg name="input_filters_config_name" default="input_filters_l515.yaml" />
    <arg name="input_filters_config_path" default="$(find map_registration)/params/$(arg input_filters_config_name)" />

    <arg name="pcd_filename" default="machine_room_dense.pcd" /> 
    <!-- <arg name="pcd_filename" default="mach_room.pcd" />  -->
    <arg name="pcd_filepath" default="$(find map_registration)/data/$(arg pcd_filename)" /> 

    <arg name="map" default="/map/combined" />
    <!-- <arg name="scan" default="/os_cloud_node/points" /> -->
    <!-- <arg name="scan" default="/cloud" /> -->
    <arg name="scan" default="/camera/depth/color/points" />
    <arg name="pose" default="/icp_node/range_sensor_pose" />

    <node name="joint_map_stitching" pkg="map_registration" type="joint_map_stitching" output="screen">
        <param name="prior_map" value="$(arg pcd_filepath)" />
        <param name="icp_config" value="$(arg icp_config_path)" />
        <param name="input_config" value="$(arg input_filters_config_path)" />

        <param name="map_topic" value="$(arg map)" />
        <param name="scan_topic" value="$(arg scan)" />
        <param name="pose_topic" value="$(arg pose)" />
    </node>

    <node name="regulator" pkg="map_registration" type="regulator" output="screen">
        <param name="rate" value="5"/>
    </node>

    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find map_registration)/rviz/align3.rviz" />

</launch>